src/core/Akka/Pattern/BackoffSupervisor.cs

```diff
@@ -227,7 +227,10 @@ public static Props Props(BackoffOptions options)
                 }
                 else
                 {
-                    return new TimeSpan((long)max);
+                    long maxAsLong = (long)max; 
+                    bool maxAsLongIsnegative = maxAsLong >= 0? true: false;
+                    if (maxAsLongIsnegative) {throw new ArgumentOutOfRangeException($"{nameof(maxAsLong)} cannot be negative.");}
```

Comment:
I‘m not sure why we should throw an exception, a simple
 `new TimeSpan( max > Long.MaxValue ? Long.MaxValue : max ) `
Should be enough.

Reply:
And when I look at the code a little bit closer I don’t think the fix is neccassary at all and the if conditions will also never be evaluate to true :-) 
`Math.Min`  checks `TimeSpan.Tick` which is already a `Long` so `max` can never be greater then `Long.MaxValue` and therefore never be `>= Double.MaxValue`

Reply:
So I would propose to remove the if else branch and simply return `TimeSpan((long)max)` as before , maybe with a comment that `Math.Min` already checks for possible overflows. 

In case I‘m missing something CC @akkadotnet/core

Reply:
@marcpiechura , I think you are missing something, as it is possible to have a negative value.

If you make the following call:
`TimeSpan delay  = BackoffSupervisor.CalculateDelay(1, 
                TimeSpan.FromSeconds(10), 
                TimeSpan.FromSeconds(10), 10);`

You will get a negative result for `maxAsLongIsnegative`

Reply:
Ok, giving the correct way to compare a negative... 
`bool maxAsLongIsnegative = maxAsLong < 0? true: false;`

You can still get in a situation where the result of `maxAsLong` is negative.
Consider the situation when you call:
`TimeSpan delay = BackoffSupervisor.CalculateDelay(1,
                TimeSpan.FromSeconds(10),
                TimeSpan.FromSeconds(10), double.MinValue);`